tests/Type/data/query-builder.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EloquentBuilder;
+
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+
+use function PHPStan\Testing\assertType;
+
+function testPluckReturnType()
+{
+    /** @var Collection<int, string> $record */
+    $record = DB::table('user')->pluck('email', 'id');
+    assertType('Illuminate\Support\Collection<int, string>', $record);
```

Comment:
```suggestion
    assertType('Illuminate\Support\Collection<array-key, string>', $record);
```

Reply:
@mad-briller  Yes exactly! I modified the code with reference to existed test cases.

By the way I found phpstan will not warn inproprite type overwritting in test case environment but it will do this in my personal repo (with same restrict level).